corehq/apps/sso/views/enterprise_admin.py

```diff
@@ -64,12 +66,16 @@ def page_context(self):
         return {
             'edit_idp_form': self.edit_enterprise_idp_form,
             'idp_slug': self.idp_slug,
+            'show_api_fields': self.show_api_fields(),
             'toggle_client_secret': (
                 self.identity_provider.protocol == IdentityProviderProtocol.OIDC
                 and self.identity_provider.client_secret
             ),
         }
 
+    def show_api_fields(self):
```

Comment:
generally for these toggle checks, `@property` is preferred

Reply:
I prefer to avoid properties if I can. Their primary purpose is to provide syntactic sugar when an object needs to conform to a specific interface. The main reason I believe they should be avoided is that they mask the complexity behind the call. When you write `object.property`, you expect no performance impact, as looking up a field on an object is trivial. However, `enabled_for_request` can ultimately result in calls to external systems (`toggle_enabled` can hit Couch, for example). This violates the law of least surprise.